web_src/js/features/comp/ImagePaste.js

```diff
@@ -1,94 +1,72 @@
 import $ from 'jquery';
-
-const {csrfToken} = window.config;
-
-async function uploadFile(file, uploadUrl) {
-  const formData = new FormData();
-  formData.append('file', file, file.name);
-
-  const res = await fetch(uploadUrl, {
-    method: 'POST',
-    headers: {'X-Csrf-Token': csrfToken},
-    body: formData,
-  });
-  return await res.json();
-}
-
-function clipboardPastedImages(e) {
-  if (!e.clipboardData) return [];
-
-  const files = [];
-  for (const item of e.clipboardData.items || []) {
-    if (!item.type || !item.type.startsWith('image/')) continue;
-    files.push(item.getAsFile());
-  }
-  return files;
-}
-
-class TextareaEditor {
-  constructor(editor) {
-    this.editor = editor;
-  }
-
-  insertPlaceholder(value) {
-    const editor = this.editor;
-    const startPos = editor.selectionStart;
-    const endPos = editor.selectionEnd;
-    editor.value = editor.value.substring(0, startPos) + value + editor.value.substring(endPos);
-    editor.selectionStart = startPos;
-    editor.selectionEnd = startPos + value.length;
-    editor.focus();
+import {getAttachedEasyMDE} from './EasyMDE.js';
+
+/**
+ *
+ * @param {*} editor
+ * @param {*} file
+ */
+export function addUploadedFileToEditor(editor, file) {
+  if (!editor && file.previewElement) {
+    editor = file.previewElement.closest('form') ? getAttachedEasyMDE(file.previewElement.closest('form').querySelector('textarea')) : getAttachedEasyMDE(file.previewElement.parentElement.parentElement.parentElement.querySelector('textarea'));
```

Comment:
This part still looks terrible, please find a better solution than `parentElement.parentElement.parentElement`, as I already said, use `el.closest` to the a common parent node and then `.querySelector` off it.